File: src/main/java/colonizer/game/Plant.java
/**
The plant object will keep track of how much hunger a plant will replace
and how grow it's food is
**/

//TODO some how makes sure all plants are all added to dictonary in map
using System;
public class plant
{
  private int foodValue = 0; // how much hunger this plant will replace
  private string plantName = "This plant has not been named!!" // name of plant
  private int howGrown = 0; // should be between 0-100%
  private Point location;
  private int xCoord; //this will hold the plants x coordinate
  private int yCoord; //this will hold the plants y coordinate
  private int value = 0; //the value of the plant at the store

  //empty constructor DO NOT USE
  public plant plant()
  {
    foodValue = 20;
    plantName = "Strange Plant";
    howGrown = 0;
  }

  //constructor
  public plant plant(int val, int price, String name, int growth, Point loc)
  {
    //sets the class varibles
    foodValue = val;
    plantName = name;
    howGrown = growth;
    location = loc;
    value = price;
  }

  //harvests the plant!!
  public plant harvest()
  {
    //creates a harvest plant, that has a gorwth of zero
    Plant harvey = new Plant(foodValue, plantName, 0);
    return harvey //returns the harvest plant so that it may be added to the inventory
  }

  //growth function for non-farmed plants
  public growBase(Plant p)
  {
    p.setHowGrown(p.getHowGrown+25);
  }

  //growth function for farmed plants
  public growFarm(Plant p)
  {
    p.setHowGrown(p.getHowGrown+50);
  }

  //NOTE for when you plant your plant in new ground. This should be done in map so
  //that the dictionary can be updated
  public plant rePlant(Plant harvey, Point location)
  {
    harvey.setLocation(location);
    return harvey;
  }

  //getters/setters
  public int getFoodValue()
  {
    return foodValue;
  }

  public setFoodValue(int v)
  {
    foodValue = v;
  }

  public int getValue()
  {
      return value;
  }

  public void setValue(int p)
  {
    value = p;
  }

  public String getPlantName()
  {
    return plantName;
  }

  public setPlantName(String name)
  {
    plantName = name;
  }

  public int getHowGrown()
  {
    return howGrown;
  }

  public setHowGrown(int growth)
  {
    howGrown = growth;
    //ensures plant growth represents a %
    if(howGrown > 100)
      howGrown = 100;
    if(howGrown < 0)
      howGrown = 0;
  }

  public Point getLocation()
  {
    return location;
  }

  public Point setLocation(Point p)
  {
    location = p;
  }
  //converts a plant into a text block
  public string toString()
  {
    string hud = "position: "+xCoord.ToString()+" by "+yCoord.ToString()+'\n'
    +plantName+" worth "+value+'\n'+"nutritional value "+foodValue+"%"+'\n'
    +howGrown+"% grown";
    return hud;
  }
}

File: src/main/java/colonizer/game/Plant.cs
/**
The plant object will keep track of how much hunger a plant will replace
and how grow it's food is
**/

using System;

namespace MissionColonizer
{
	public class Plant
	{
		private int foodValue = 0;
		private string plantName = "This plant has not yet been named";
		private int howGrown = 0;
		private bool isFarmed = false; //TODO getters & setters
		private int price = 0; //value in store

		public Plant()
		{
			foodValue = 0;
			howGrown = 0;
			price = 10;
			plantName = "Strange Plant";
		}

		public Plant(int food, string name, int grown, int p)
		{
			foodValue = food;
			plantName = name;
			howGrown = grown;
			price = p;
		}

		public Plant harvest()
		{
			return this;
		}

		// Increase how much the plant has grown by base num (25), more if farmed
		public void growBase()
		{
			howGrown = howGrown + 25; //base growth for all plants
			if(isFarmed) //if farmed
				howGrown = howGrown + 25; //grow by a total of 50

			// check to make sure it hasn't overgrown or undergrown
			if (howGrown > 100)
				howGrown = 100;
			else if (howGrown <= 0)
				howGrown = 0;
		}

		// Return how much this plant has grown
		public int getHowGrown()
		{
			return howGrown;
		}

		// Set how much this plant has grown
		public void setHowGrown(int growth)
		{
			howGrown = growth;

			// check to make sure it hasn't overgrown or undergrown
			if (howGrown > 100)
				howGrown = 100;
			else if (howGrown <= 0)
				howGrown = 0;
		}

		// Reset the howGrown value to new value
		public void rePlant(int newHowGrown)
		{
			howGrown = newHowGrown;
			isFarmed = true;
			//TODO Might need some more stuff here
			//NOTE does a plant know where it is? Does it need to?
		}

		// Return the food value of this plant
		public int getFoodValue()
		{
			return foodValue;
		}

		// Set a new food value for this plant
		public void setFoodValue(int val)
		{
			foodValue = val;
		}

		// Return the name of this plant
		public string getPlantName()
		{
			return plantName;
		}

		// Set the name of this plant
		public void setPlantName(string name)
		{
			plantName = name;
		}

		//converts a plant into a text block
	  public string toString()
	  {
	    string info = plantName+" worth "+value+"Cr\n"
			+"nutritional value "+foodValue+"%"+'\n';
	    return info;
	  }
	}
}
